apps/payments/next/providers/AppLocalizationProvider.tsx

```diff
@@ -0,0 +1,56 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use client';
+
+import React, { createContext, useContext, useState, useEffect } from 'react';
+import { ReactLocalization } from '@fluent/react';
+import { useParams } from 'next/navigation';
+import { getLocaleFromRequest } from '@fxa/shared/l10n';
+import {
+  LocalizerClient,
+  LocalizerBindingsClient,
+} from '@fxa/shared/l10n/client';
+
+const AppLocalizationContext = createContext<ReactLocalization | undefined>(
+  undefined
+);
+
+export function AppLocalizationProvider({
+  children,
+}: {
+  children: React.ReactNode;
+}) {
+  const [l10n, setL10n] = useState<ReactLocalization>();
+  const [isLocalized, setIsLocalized] = useState<boolean>(true);
+
+  const params = useParams();
+  const locale = getLocaleFromRequest(params, null);
+
+  useEffect(() => {
+    const setLocalization = async () => {
+      const bindings = new LocalizerBindingsClient();
+      bindings.fetchResource(
+        `${bindings.opts.translations.basePath}/${locale}/main.ftl`
+      );
+      const localizerClient = new LocalizerClient(bindings);
+      const l10nPromise = await localizerClient.setupReactLocalization(locale);
+      await Promise.resolve(l10nPromise).then((res) => setL10n(res.l10n));
+
+      setIsLocalized(false);
+    };
+
+    isLocalized && setLocalization();
+  }, [locale, l10n, isLocalized]);
+
+  return (
+    <AppLocalizationContext.Provider value={l10n}>
```

Comment:
So I switched the logic into RootLayout and passed the ReactLocalization instance `l10n` into `<LocalizationProvider>` from the fluent/react library there. I did try doing this in a separate component, provider, and function, but this seemed the most straightforward to me and I was doing it as a test since I'm having issues with my local. I don't know if this is what you were thinking. I am sort of working blind since I can't get my page to load.

I particularly looked at https://vercel.com/guides/react-context-state-management-nextjs#rendering-third-party-context-providers-in-server-components and https://github.com/projectfluent/fluent.js/wiki/LocalizationProvider, as well as several other resources that all slightly did it differently.

Is this what you are thinking? I would like to be on the same page of how you were thinking this PR would look.

Reply:
Yup this is close. 

The logic in the root layout looks mostly correct and should be moved into this `AppLocalizationProvider` component. The reason it shouldn't be in the root layout, is that ideally the root layout should remain a server component. (i.e. `use client` shouldn't be added to the root layout) The steps mentioned in the vercel article provides the work around to adding the provider, so that context is available throughout the app, while still using Server Components.